ocis-pkg/service/debug/service.go

```diff
@@ -46,8 +48,16 @@ func NewService(opts ...Option) *http.Server {
 		mux.Handle("/debug", h)
 	}
 
+	baseCtx := context.Background()
+	if dopts.Context != nil {
+		baseCtx = dopts.Context
+	}
```

Comment:
Wouldn't it make more sense to flip this? Init with `dopts. Context` and if that is nil use context.Background?

Reply:
true enough. fixed.